homeassistant/components/godice/__init__.py

```diff
@@ -0,0 +1,76 @@
+"""The godice integration."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+import logging
+
+import godice
+
+from homeassistant.components import bluetooth
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ADDRESS, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import CONF_SHELL
+
+PLATFORMS = [Platform.SENSOR]
+_LOGGER = logging.getLogger(__name__)
+
+type DiceConfigEntry = ConfigEntry[DiceData]
+
+
+@dataclass
+class DiceData:
+    """Runtime data to interact with GoDice."""
+
+    dice: godice.Dice
+    disconnected_by_request_flag: bool
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: DiceConfigEntry) -> bool:
+    """Set up godice from a config entry."""
+    _LOGGER.debug("Setup started")
+
+    def on_disconnect_callback(_ble_data):
+        _LOGGER.debug("on_disconnect_callback called")
+        is_disconnected_by_request = (
+            config_entry.runtime_data.disconnected_by_request_flag
+        )
+        if not is_disconnected_by_request:
+            hass.create_task(hass.config_entries.async_reload(config_entry.entry_id))
```

Comment:
This looks like it will likely run in the event loop but is using the sync api (`create_task`)